plugins/org.yakindu.sct.doc.user/src/user-guide/embedded_systems_integration_guide.textile

```diff
@@ -0,0 +1,281 @@
+* In Events 
+* Timed Events
+* Out Events
+
+h3(#main-function). Main Function
+
+The main function could be realized as following: First, the hardware will be initialized. Then the timers will take into account. After this the state machine will be initialized and entered. 
+At this point the state machine will be called as long as the state machine is not final. Note: If there is no final state defined the behavior is equals a while(true) loop. At first all in events will be raised. After this the timer will be updated, which internally raises time events. The behavior equals an in events. Then the runCycle will be called, which handles the logical execution of the state machine. This only needed, if the execution semantics is cycle based and must only be executed based on the elapsed time, e.g. 200 ms for a CycleBased(200) state machine. After proceeding, the out events will be handled. The last step, if wanted, is setting the microcontroller into a sleep mode.
+
+bc.. 
+void main() {
+	// Initialization
+	
+	hardwareInit();
+	
+	if(hasTimeEvents) {
+		timerInit();
+	}
+	
+	statemachine_init();
+	statemachine_enter();
+
+	// While loop
+	while(!statemachine_isFinal()) {
+		if(hasInEvents) {
+			handleInEvents();
+		}
+		
+		if(hasTimeEvents) {
+			handleTimer();
+		}
+		
+		if(isCycleBased && elapsedTime >= cyclePeriod) {
+			elapsedTime = 0;
+			statemachine_runCycle();
+		}
+		
+		if(hasOutEvents) {
+			handleOutEvents();
+		}
+		
+		if(usingSleepMode) {
+			goToSleep()
+			// wait for Interrupt
+		}
+	}
+}
+p. 
+
+h3(#handle-in-events). Handle In Events 
+
+For now on, there are two different ways of how the events should be interact on the embedded system. One design pattern is polling. The other one is the usage of interrupts.
+
+A design using polling cyclic updates the status of the inputs and raise the according event.
+
+bc.. 
+void handleInEvents() {
+	if(readGPIO(1)) {
+		statemachine_raise_inEvent1();
+	}
+	if(readGPIO(2)) {
+		statemachine_raise_inEvent2();
+	}
+}
+p. 
+
+As already described in the Interrupt section, a design using interrupts recommends bool flags as storage of the event.
+
+bc.. 
+void handleInEvents() {
+	if(inEvent1) {
+		statemachine_raise_inEvent1();
+		inEvent1 = false;
+	}
+	if(inEvent2) {
+		statemachine_raise_inEvent2(inEvent2Value);
+		inEvent2 = false;
+	}
+}
+
+Sensor1_ISR{
+	inEvent1 = true;
+}
+
+Sensor2_ISR{
+	inEvent2 = true;
+	inEvent2Value = readSensor2();
+}
+p. 
+
+h3(#handle-timer). Handle Timer
+
+Handling the timer is more comprehensive. Each time event of the state machine must be updated and handled, because in detail there are nothing else than a in event. This means that for every timed trigger there must be one in event. Therefore, a timer service is provided in the examples. This timer service must be updated with the elapsed time since the its last call, which leads to the next implementation point: Determining the time.
```

Comment:
```suggestion
Handling the timer is more comprehensive. Each time event of the state machine must be updated and handled, because in detail they are nothing else than an in event. Therefore, a timer service is provided in the examples. This timer service must be updated with the elapsed time since its last call, which leads to the next implementation point: Determining the time.
```